model/resolver/validation.go

```diff
@@ -541,37 +539,46 @@ func validateColocatedContainerVolumeShares(roleManifest *model.RoleManifest) va
 				if volume.Type == model.VolumeTypeEmptyDir {
 					emptyDirVolumesTags = append(emptyDirVolumesTags, volume.Tag)
 					emptyDirVolumesPath[volume.Tag] = volume.Path
+
+					// Initialize counters of volume tags of the main instance group
 					emptyDirVolumesCount[volume.Tag] = 0
 				}
 			}
 
-			for _, colocatedRole := range instanceGroup.GetColocatedRoles() {
-				for _, volume := range colocatedRole.Run.Volumes {
+			for _, colocatedInstanceGroup := range instanceGroup.GetColocatedRoles() {
+				for _, volume := range colocatedInstanceGroup.Run.Volumes {
 					if volume.Type == model.VolumeTypeEmptyDir {
+						// Initialize an additional volume tag in case the colocated
+						// containers brings a new tag into the game.
 						if _, ok := emptyDirVolumesCount[volume.Tag]; !ok {
 							emptyDirVolumesCount[volume.Tag] = 0
 						}
 
+						// Count the usage of a volume (by tag) of a colocated container
 						emptyDirVolumesCount[volume.Tag]++
 
-						if path, ok := emptyDirVolumesPath[volume.Tag]; ok {
-							if path != volume.Path {
-								// Same tag, but different paths
-								allErrs = append(allErrs, validation.Invalid(
-									fmt.Sprintf("instance_group[%s]", colocatedRole.Name),
-									volume.Path,
-									fmt.Sprintf("colocated instance group specifies a shared volume with tag %s, which path does not match the path of the main instance group shared volume with the same tag", volume.Tag)))
-							}
+						// Validate that volumes with the same tag in both the main instance group
+						// and the colocated container have the same path configured. The same tag
+						// but a different path results in a validation error.
+						if path, ok := emptyDirVolumesPath[volume.Tag]; ok && path != volume.Path {
+							// Same tag, but different paths
+							allErrs = append(allErrs, validation.Invalid(
+								fmt.Sprintf("instance_group[%s]", colocatedInstanceGroup.Name),
+								volume.Path,
+								fmt.Sprintf("colocated instance group specifies a shared volume with tag %s, which path does not match the path of the main instance group shared volume with the same tag", volume.Tag)))
```

Comment:
Not really part of your PR, but: shouldn't `instanceGroup.Name` be shown somewhere?